rfcs/0174-improve-terminology.md

```diff
@@ -0,0 +1,160 @@
+- Rename "derivation" to "derivation file", keeping the current definition. Define "derivation" as "an old term that is no longer used, which usually means derivation files"
+- Treat every other usage of the word "derivation" as **incorrect documentation**
+- Remove the word "instantiate". For all intents and purposes, "instantiation" is just putting a derivation file into Nix store; this is an implementation detail
+- Replace the words "realise", "realisation" with "install", "installation". Nix operates on installables, and it installs them; there's no need for an arcane term
+- Rename `lib.mkDerivation` function to `lib.mkPackage`. While mkDerivation can be used for defining FOD or IFD, it usually isn't, so we dodge the edge cases where "derivation is not technically a package"
+- Rename `builtins.derivation` to `builtins.drvFile`
+
+## Projects
+
+There are a few issues present:
+
+- "Nix language" and "Nix package manager" share the name "Nix"
+- Further, "Nix package manager" isn't an official term that exists. We just have "Nix", which is a package manager, and the distributed package includes a daemon and a CLI to interact with it
+- Nixpkgs repository is very awkwardly named. When spelled out, "Nixpkgs" sounds the same as "Nix packages". This makes it hard to introduce people to Nix with audio presentation
+- There isn't consistent naming for the community and its activities. Looking at the [community page](https://nixos.org/community/), we can see the following used:
+  - The website itself is under domain nixos.org (implying that related organization/community is NixOS)
+  - "Nix community" is used to describe the whole community
+  - The official Matrix space is also hosted under nixos.org
+  - Mastodon and Twitter, which are described as "official spaces", are also tagged under NixOS
+  - Stackoverflow link to "official space" looks for tags: "Nix", "Nixpkgs", "NixOS", "NixOps"
+  - NixCon is named after Nix, not NixOS
+  - The link under "RFC process" leads to the official organization on Github, which is called NixOS
+  - Under the "foundation board" entry, "NixOS project" is mentioned
+  - Under the "security team" entry, "NixOS community" is mentioned
+  - Under the "infrastructure team", both "Nix community" and "NixOS community" are mentioned; likewise for "NixCon team"
+  - Under the "marketing team", "Nix user base" is mentioned
+  - Under the "CUDA team", both "Nix" and "NixOS" are mentioned
+  - There's a "Nix team", which is responsible for Nix package manager specifically; makes sense
+  - There's a "Nix formatting team", which creates a standard format for "Nix". It is ambiguous if the team creates a standard format for Nix package manager or for Nix DSL
+  - NixOS Foundation is mentioned, which once again suggests that the project is NixOS
+
+Here's the solution proposed:
+
+- Give an official name to Nix DSL. One of the proposed names is just "Nix DSL". A very popular alternative is "NixLang", similar to Go/Golang
```

Comment:
Right. Nix manual even has a [section](https://nixos.org/manual/nix/stable/language/) dedicated to it. From what I remember over the years and from my recent re-reads of manuals (which were partial re-reads), the term doesn't seem to be consistently used.

What I'm trying to say is, "Nix language" is a phrase that sees use, but it's not officially enforced as much as it should be. The downside of saying "Nix language" is that it's a bit of a mouthful; I don't mind it otherwise, personally.

The argument for not using "Nix DSL" is valid. I'll add a section that lists the possible names and their upsides and downsides a little bit later, so that the naming discussions can be localized there.

Reply:
Calling it Nix Language requires very little change while Nix DSL is a complete new name which probably creates more confusion what it is about than it resolves.